src/OnlineSales/Controllers/BaseControllerWithImport.cs

```diff
@@ -23,15 +23,91 @@ public BaseControllerWithImport(ApiDbContext dbContext, IMapper mapper, IOptions
     {
     }
 
+    public int ImportBatchSize { get; set; } = 5;
+
+    public long FileSizeInMB { get; set; } = 5;
+
     [HttpPost]
     [Route("import")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status422UnprocessableEntity)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status500InternalServerError)]
     public virtual async Task<ActionResult> Import([FromBody] List<TI> records)
     {
-        // TODO: Add import logic here
+        ValidateUploadFileSize();
```

Comment:
@gsplakmal I have two concern at this line of code:

1. ***FileSize is a wrong name because we do not expect any file to be sent. When you say a "file" it usually means we expect a client to send request body as a multipart/form-data while we really expect body as a text with Content-Type = text/csv or text/json.
2. It is a wrong place to verify size because at this point the "records" collection would have already been fully uploaded to the server and deserialised into the RAM. The whole point of size validation is to prevent .NET pipeline to deserialise very big input file into the memory because it could cause out of memory exception and kill the app. We must prevent the full body from being uploaded to the server and throw an exception much earlier in the pipeline, even before the full upload happens (the client usually sends Content-Length header which you can check before full body data is uploaded). Likely, .NET already has the default solution for request length validation and it even has some default limits. There are different way how it could be implemented using standard approach, please see more details here: https://www.talkingdotnet.com/how-to-increase-file-upload-size-asp-net-core/ - note, this article describes .NET Core 2.0, but I believe it should be very similar in .NET 7.